bennu-portal/src/main/java/org/fenixedu/bennu/portal/domain/MenuContainer.java

```diff
@@ -160,7 +164,15 @@ public Set<MenuItem> getOrderedChild() {
      *         The User's Menu as a Stream
      */
     public Stream<MenuItem> getUserMenuStream() {
-        return getChildSet().stream().filter((item) -> item.isItemVisible() && item.isItemAvailableForCurrentUser()).sorted();
+        return getUserMenuStream(Authenticate.getUser());
+    }
+
+    public Stream<MenuItem> getUserMenuStream(User user) {
```

Comment:
Why we have 2 methods one for user other for ApplicationUser?
Why one not call other?

Reply:
There are two getUserMenuStream methods, one for Bennu users and another one for ApplicationUsers, for 2 reasons:
- Originally there was only the getUserMenuStream() method which did the same as the getUserMenuStream(User) method but using the Bennu user obtained by calling Authenticate.getUser(), meaning that in order to guarantee that the getUserMennuStream() had the same behaviour after splitting it into two methods, the other getUserMenuStream had to receive a Bennu user obtained by calling Authenticate.getUser()
- The getUserMenuStream(ApplicationUser) was created to simplify the getAvailableChilds(ApplicationUser) method, which now only has to call the getUserMenuStream method directly using its parameter, without having to check if it's null, and if it's not, get the corresponding Bennu user

To eliminate one of the getUserMenuStream methods, having one call the other would work (with getUserMenuStream(AppUser) calling getUserMenuStream(User) or vice-versa), but it would lead to yet another null check (there is already a null check on the isAvailableForUser(ApplicationUser) method for this reason) on the the method calling the other, since in order to convert Bennu users to ApplicationUsers and vice-versa, it is necessary to check if they are null before getting their username, meaning that it would be preferable to simply eliminate the getUserMenuStream(AppUser) method and call the getUserMenuStream(User) method directly from the getAvailableChilds method.

Another option would be to eliminate the getUserMenuStream(User) method and call the getUserMenuStream(ApplicationUser.currentUser) in the getUserMenuStream() method, my only concern is that calling ApplicationUser.currentUser may not always yield the same result as calling Authenticate.getUser

For these reasons I deicded to eliminate the getUserMenuStream(AppUser) method and call the gettUserMenuStream(User) method in getAvailbleChilds(AppUser), checking if the passed appUser is null before getting the corresponding Bennu user